src/EditorFeatures/Core/Implementation/PasteTracking/PasteTrackingPasteCommandHandlercs.cs

```diff
@@ -17,8 +17,7 @@ namespace Microsoft.CodeAnalysis.PasteTracking
     [Export(typeof(VSCommanding.ICommandHandler))]
     [ContentType(ContentTypeNames.RoslynContentType)]
     [Name(PredefinedCommandHandlerNames.PasteTrackingPaste)]
-    [Order(After = PredefinedCommandHandlerNames.FormatDocument)]
-    [Order(Before = PredefinedCommandHandlerNames.Completion)]
+    [Order(Before = PredefinedCommandHandlerNames.FormatDocument)]
```

Comment:
note: the above seems wrong.  it seems like we would want to call into formatting from us, so that we can accurately track the entire before/after span of the change.  **however**, what we would also want to do is *disconnect* from listening to buffer changes while this is happening.

Reply:
@CyrusNajmabadi That is what this change enables. In the `ExecuteCommand` method the `nextCommandHandler` will FormatDocument or or another handler that chains into FormatDocument. Since we invoke `nextCommandHandler`  we are calling into formatting and getting to track the span of inserted and formatted code. No need to disconnect in this case since we are allowing the formatting to finish before registering the pasted span.

Reply:
oh... duh.  sorry, i reversed the diff in my head.  Thanks much! 👍